flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/aggregate/TimeSortProcessFunctionTest.scala

```diff
@@ -0,0 +1,256 @@
+import org.apache.flink.types.Row
+import org.junit.Test
+import org.apache.flink.table.runtime.aggregate.ProcTimeSortProcessFunction
+import org.apache.flink.table.runtime.aggregate.RowTimeSortProcessFunction
+import org.apache.flink.table.runtime.aggregate.TimeSortProcessFunctionTest._
+import org.apache.flink.api.java.typeutils.runtime.RowComparator
+import org.apache.flink.api.common.typeutils.TypeSerializer
+import org.apache.flink.api.common.typeutils.TypeComparator
+import org.apache.flink.table.runtime.types.{CRow, CRowTypeInfo}
+import org.apache.flink.streaming.api.TimeCharacteristic
+
+class TimeSortProcessFunctionTest{
+
+  
+  @Test
+  def testSortProcTimeHarnessPartitioned(): Unit = {
+    
+    val rT =  new RowTypeInfo(Array[TypeInformation[_]](
+      INT_TYPE_INFO,
+      LONG_TYPE_INFO,
+      INT_TYPE_INFO,
+      STRING_TYPE_INFO,
+      LONG_TYPE_INFO),
+      Array("a","b","c","d","e"))
+    
+    val rTA =  new RowTypeInfo(Array[TypeInformation[_]](
+     LONG_TYPE_INFO), Array("count"))
+    val indexes = Array(1,2)
+      
+    val fieldComps = Array[TypeComparator[AnyRef]](
+      LONG_TYPE_INFO.createComparator(true, null).asInstanceOf[TypeComparator[AnyRef]],
+      INT_TYPE_INFO.createComparator(false, null).asInstanceOf[TypeComparator[AnyRef]] )
+    val booleanOrders = Array(true, false)    
+    
+
+    val rowComp = new RowComparator(
+      rT.getTotalFields,
+      indexes,
+      fieldComps,
+      new Array[TypeSerializer[AnyRef]](0), //used only for serialized comparisons
+      booleanOrders)
+    
+    val collectionRowComparator = new CollectionRowComparator(rowComp)
+    
+    val inputCRowType = CRowTypeInfo(rT)
+    
+    val processFunction = new KeyedProcessOperator[Integer,CRow,CRow](
+      new ProcTimeSortProcessFunction(
+        inputCRowType,
+        collectionRowComparator))
+  
+   val testHarness = new KeyedOneInputStreamOperatorTestHarness[Integer,CRow,CRow](
+      processFunction, 
+      new TupleRowSelector(0), 
+      BasicTypeInfo.INT_TYPE_INFO)
+    
+   testHarness.open();
+
+   testHarness.setProcessingTime(3)
+
+      // timestamp is ignored in processing time
+    testHarness.processElement(new StreamRecord(new CRow(
+      Row.of(1: JInt, 11L: JLong, 1: JInt, "aaa", 11L: JLong), true), 1001))
+    testHarness.processElement(new StreamRecord(new CRow(
+        Row.of(1: JInt, 12L: JLong, 1: JInt, "aaa", 11L: JLong), true), 2002))
+    testHarness.processElement(new StreamRecord(new CRow(
+        Row.of(1: JInt, 12L: JLong, 2: JInt, "aaa", 11L: JLong), true), 2003))
+    testHarness.processElement(new StreamRecord(new CRow(
+        Row.of(1: JInt, 12L: JLong, 0: JInt, "aaa", 11L: JLong), true), 2004))
+    testHarness.processElement(new StreamRecord(new CRow(
+        Row.of(1: JInt, 10L: JLong, 0: JInt, "aaa", 11L: JLong), true), 2006))
+
+    //move the timestamp to ensure the execution
+    testHarness.setProcessingTime(1005)
```

Comment:
Please run at least two sorts to ensure that all intermediate state is correctly reset.
Also verify the state size to ensure that the operator is not leaking.